colossus/src/main/scala/colossus/protocols/http/Header.scala

```diff
@@ -196,9 +220,9 @@ class HttpHeaders(private val headers: JList[HttpHeader]) {
    */
   def contentLength: Option[Int] = firstValue(HttpHeaders.ContentLength).map{_.toInt}
 
-  def transferEncoding : TransferEncoding = firstValue(HttpHeaders.TransferEncoding).flatMap(TransferEncoding.unapply).getOrElse(TransferEncoding.Identity)
+  def transferEncoding : TransferEncoding = firstValue(HttpHeaders.TransferEncoding).map(TransferEncoding(_)).getOrElse(TransferEncoding.Identity)
 
-  def connection: Option[Connection] = firstValue(HttpHeaders.Connection).flatMap(Connection.unapply)
+  def connection: Option[Connection] = firstValue(HttpHeaders.Connection).map(Connection(_))
```

Comment:
one idea is to have firstValue take in those map functions as a parameter so that they can be directly applied to the collectFirst function.